pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -83,8 +83,15 @@ func parseControllerCerts(ctx *zedagentContext, contents []byte) (changed bool,
 				break
 			}
 		}
-		if !found {
-			log.Functionf("parseControllerCerts: deleting %s", config.Key())
+		// XXX move this inside if statement
+		// Is the ControllerCert in use?
+		if lookupCipherContextByCCH(ctx.getconfigCtx, configHash) != nil {
```

Comment:
Should we put it under
```
if !found {
```
so that we do the extra check only for the certs that are not in the latest config and hence are up to be removed?

Reply:
Cleanup up the XXX printfs and restructured this.